MediaBrowser.Providers/MediaInfo/AudioFileProber.cs

```diff
@@ -355,6 +355,28 @@ private async Task FetchDataFromTags(Audio audio, Model.MediaInfo.MediaInfo medi
                 }
             }
 
+            if (options.ReplaceAllMetadata || !audio.TryGetProviderId(MetadataProvider.MusicBrainzRecording, out _))
+            {
+                if ((track.AdditionalFields.TryGetValue("MUSICBRAINZ_TRACKID", out var recordingMbId)
+                     || track.AdditionalFields.TryGetValue("MusicBrainz Track Id", out recordingMbId))
+                    && !string.IsNullOrEmpty(recordingMbId))
+                {
+                    audio.TrySetProviderId(MetadataProvider.MusicBrainzTrack, recordingMbId);
+                }
+                else if (track.AdditionalFields.TryGetValue("UFID", out var ufIdValue) && !string.IsNullOrEmpty(ufIdValue))
+                {
+                    // If tagged with MB Picard, the format is 'http://musicbrainz.org\0<recording MBID>
+                    if (ufIdValue.Contains("musicbrainz.org", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var ufidParts = ufIdValue.Split('\0');
+                        if (ufidParts.Length > 1)
+                        {
+                            audio.TrySetProviderId(MetadataProvider.MusicBrainzRecording, ufidParts[1]);
+                        }
```

Comment:
```suggestion
                        audio.TrySetProviderId(MetadataProvider.MusicBrainzRecording, ufIdValue.AsSpan().RightPart('\0').ToString());
```

Reply:
Thanks, added in https://github.com/jellyfin/jellyfin/pull/12173/commits/33fe27be8c4f0d2226a3ad4d05670a4d0592d89d